### PR TITLE
Add soft block waffle switch

### PR DESCRIPTION
**Pull request by KevinMind**
Fixes: mozilla/addons#15162

### Description

Add a soft block waffle switch to allow control of creating soft block stash/filter 

### Context

Once we start uploading soft block filter/stash FX will start using it immediately. So we should be able to turn off soft blocking in the event that something goes wrong in initial release

### Testing

This switch will be used in other PRs so testing it there is what counts. THis just creates it in the database.

### Checklist

- [ ] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [ ] Successfully verified the change locally.
- [ ] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add before and after screenshots (Only for changes that impact the UI).
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

### CODE REVIEW

File: src/olympia/blocklist/migrations/0038_auto_20241112_1606.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('blocklist', '0037_alter_blockversion_block_type_and_more'),
+    ]
+
+    operations = [
+        CreateWaffleSwitch('mlbf-soft-blocks-enabled'),
```
Comment (willdurand) [inline src/olympia/blocklist/migrations/0038_auto_20241112_1606.py:15]:
```suggestion
        CreateWaffleSwitch('soft-blocking-enabled'),
```

**Review by willdurand — COMMENTED** [review]
I guess I am fine with a new switch but why not using it somewhere? It could be added to one of the other patches in review IMO.

File: src/olympia/blocklist/migrations/0038_auto_20241112_1606.py (hunk 1 above)
Comment (KevinMind) [reply src/olympia/blocklist/migrations/0038_auto_20241112_1606.py:15]: @willdurand it's used in both open PRs currently.

File: src/olympia/blocklist/migrations/0038_auto_20241112_1606.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('blocklist', '0037_alter_blockversion_block_type_and_more'),
+    ]
+
+    operations = [
+        CreateWaffleSwitch('soft-blocking-enabled'),
```
Comment (diox) [inline src/olympia/blocklist/migrations/0038_auto_20241112_1606.py:15]: I would call that `enable-soft-blocking` for consistency with _most_ other waffle switches governing features that we already have.

**Review by diox — APPROVED** [review]
Please change the name.

I would have added this to the bloomfilter PR and made the stashes one wait for the bloomfilter PR, but that works too I guess... Just a little bit confusing/misleading to introduce a switch that currently does nothing, but it's fine.

**Comment by willdurand** [comment]
```
E   django.core.management.base.CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0038_auto_20241112_1606, 0038_tweak_blocksubmission_choices in blocklist).
```

**Comment by diox** [comment]
Yeah I merged 2a1eb8b1c55ba8d08fbc8ed769a303ad8726091e which added `blocklist` migration `0038`, this needs a rebase and bumping the number as well.

**Pull request merged**